lib/webpack.js

```diff
@@ -132,3 +132,25 @@ exportPlugins(exports.node = {}, {
 exportPlugins(exports.debug = {}, {
 	"ProfilingPlugin": () => require("./debug/ProfilingPlugin"),
 });
+
+const defineMissingPluginError = (pluginName, errorMessage) => {
+	Reflect.defineProperty(exports.optimize, pluginName, {
+		configurable: false,
+		enumerable: true,
+		get() {
+			throw new Error(`RemovedPluginError: ${errorMessage}`);
+		}
+	});
+};
+
+// TODO remove in webpack 5
+defineMissingPluginError(
```

Comment:
pass `exports.optimize` as first argument, so we could define missing plugins for other categories too.